contracts/src/v0.8/functions/dev/v1_X/FunctionsBilling.sol

```diff
@@ -124,6 +142,27 @@ abstract contract FunctionsBilling is Routable, IFunctionsBilling {
     return SafeCast.toUint96((1e18 * amountWei) / getWeiPerUnitLink());
   }
 
+  /// @inheritdoc IFunctionsBilling
+  function getUsdPerUnitLink() public view returns (uint256, uint8) {
+    FunctionsBillingConfig memory config = s_config;
+    (, int256 usdPerUnitLink, , uint256 timestamp, ) = s_linkToUsdFeed.latestRoundData();
+    // solhint-disable-next-line not-rely-on-time
+    if (config.feedStalenessSeconds < block.timestamp - timestamp && config.feedStalenessSeconds > 0) {
+      return (config.fallbackUsdPerUnitLink, config.fallbackUsdPerUnitLinkDecimals);
+    }
+    if (usdPerUnitLink <= 0) {
+      revert InvalidUsdLinkPrice(usdPerUnitLink);
+    }
+    return (uint256(usdPerUnitLink), s_linkToUsdFeed.decimals());
+  }
+
+  function _getJuelsFromUsd(uint256 amountUsd) private view returns (uint96) {
```

Comment:
I need to work through this math by hand to verify.

Reply:
I doubled checked. I think this is good.

Reply:
We also have tests that triple and quadruple check this, right?

Reply:
The tests are in this, yes. They should also be reviewed.